src/Form/Extension/ChoiceTypeExtension.php

```diff
@@ -13,38 +13,35 @@
 
 namespace Sonata\AdminBundle\Form\Extension;
 
-use Symfony\Component\Form\AbstractTypeExtension;
 use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
-use Symfony\Component\Form\FormInterface;
-use Symfony\Component\Form\FormView;
-use Symfony\Component\OptionsResolver\OptionsResolver;
-
-/**
- * @final since sonata-project/admin-bundle 3.52
- *
- * @author Amine Zaghdoudi <amine.zaghdoudi@ekino.com>
- */
-class ChoiceTypeExtension extends AbstractTypeExtension
-{
-    public function configureOptions(OptionsResolver $resolver)
+use Symfony\Component\Form\FormTypeExtensionInterface;
+
+// NEXT_MAJOR: Remove the "else" part, copy all methods from BaseChoiceTypeExtension in this class and
```

Comment:
The cost of maintaining this for all the types we have it is simply too much, im -1 for this type of solution. And it is not for a short of time, it will take at least 2 years to remove support for sf 4.4

Reply:
It's only in the form type extensions (we have 3), I don't know if there is other solution rather than dropping support for Symfony 4.4 or adding a minimal BC-break (those classes were marked as final 1 year ago).

Reply:
I prefer to do a bc break like we did in the past with other minimal things on a class marked as final in next major.

Reply:
Since the classes are marked `@final` I would just add the typehint? Whoever still extends those classes is not covered by any BC promises, or?

Reply:
@dmaicher we did not introduce the `@final` annotations in a new major I'm afraid so I think it will be a BC-break. These annotations are here to warn people that we are going to use the `final` keyword in the next major release. That being said, we have made such minor BC-breaks in the past.